### PR TITLE
chore: abstract priceToTick loop to helper func

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #3985 

## What is the purpose of the change

When computing PriceToTick, we previously looped through exponents (incrementing or decrementing depending on if the desired price is above or below 1) until we reached the first exponent that is greater than the desired price:

https://github.com/osmosis-labs/osmosis/blob/00c430bab18e8ac9d4776180d677ca3b65aaaec5/x/concentrated-liquidity/internal/math/tick.go#L124-L143

This PR abstracts this logic into a helper function with tests, as requested in the below discussion:

https://github.com/osmosis-labs/osmosis/pull/3874#discussion_r1066576987

## Brief Changelog

- Abstracts priceToTick loop into `CalculatePriceAndTicksPassed` function
- Adds tests for this function

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/concentrated-liquidity/internal/math/tick.go [hunk 1]
```diff
@@ -188,3 +155,48 @@ func GetMinAndMaxTicksFromExponentAtPriceOneInternal(exponentAtPriceOne sdk.Int)
 	maxTick = sdkThirtyEightDec.Mul(geometricExponentIncrementDistanceInTicks).TruncateInt64()
 	return minTick, maxTick
 }
+
+// calculatePriceAndTicksPassed takes in a price and an exponentAtPriceOne, and returns the currentPrice, ticksPassed, and currentAdditiveIncrementInTicks.
+// The function uses the geometricExponentIncrementDistanceInTicks formula to determine the number of ticks passed and the current additive increment in ticks.
+// If the price is greater than 1, the function increments the exponentAtCurrentTick until the currentPrice is greater than the input price.
+// If the price is less than 1, the function decrements the exponentAtCurrentTick until the currentPrice is less than the input price.
+func CalculatePriceAndTicksPassed(price sdk.Dec, exponentAtPriceOne sdk.Int) (currentPrice sdk.Dec, ticksPassed sdk.Int, currentAdditiveIncrementInTicks osmomath.BigDec) {
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/internal/math/tick.go:163]: This should probably be private right?

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM, left minor comment

File: x/concentrated-liquidity/internal/math/tick.go [hunk 2]
```diff
@@ -188,3 +155,48 @@ func GetMinAndMaxTicksFromExponentAtPriceOneInternal(exponentAtPriceOne sdk.Int)
...
+
+	// Initialize the current price to 1, the current precision to exponentAtPriceOne, and the number of ticks passed to 0
+	currentPrice = sdk.OneDec()
+	ticksPassed = sdk.ZeroInt()
+	exponentAtCurrentTick := exponentAtPriceOne
+
+	// Set the currentAdditiveIncrementInTicks to the exponentAtPriceOne
+	currentAdditiveIncrementInTicks = powTenBigDec(exponentAtPriceOne)
+
+	// Now, we loop through the exponentAtCurrentTicks until we have passed the price
```
Comment (mattverse) [inline x/concentrated-liquidity/internal/math/tick.go:176]: Confirmed that we don't have logic changes here

**Review by mattverse — APPROVED** [review]
LGTM, I personally dont think we need more test cases for this PR

**Pull request merged**